lib/ansible/module_utils/facts/collector.py

```diff
@@ -356,7 +359,11 @@ def collector_classes_from_gather_subset(all_collector_classes=None,
 
     minimal_gather_subset = minimal_gather_subset or frozenset()
 
-    platform_info = platform_info or {'system': platform.system()}
+    if connection:
+        capabilities = json.loads(connection.get_capabilities())
```

Comment:
Seems like we need some validation of the object type for "connection". It's also not clear where "capabilities" are set from this patch alone. Is that config / variables? Is there some sort of input validation happening elsewhere that warns when something is missing or wrong?

Reply:
fixed by https://github.com/ansible/ansible/commit/9118c5ff7633f732affb0bd5b41dcd0359f677d7